src/Cloud4RPi.cpp

```diff
@@ -1,23 +1,30 @@
 #include "Cloud4RPi.h"
 
+const int JSON_BUFFER_SIZE = 500;
+
 Cloud4RPi::Cloud4RPi(const String &_deviceToken, const String &_server, int _port) :
     deviceToken(_deviceToken),
     server(_server),
     port(_port),
-    mqttClient(NULL) {
-      // Empty
-}
-
-void Cloud4RPi::begin(Client& _client) {
-  mqttClient = new PubSubClient(_client);
-  mqttClient->setServer(server.c_str(), port);
+    mqttClient(NULL),
+    jsonBufferSize(JSON_BUFFER_SIZE),
+    variables(new C4RVariableStorage()) {
 }
 
 Cloud4RPi::~Cloud4RPi() {
-  if (mqttClient == NULL) {
+  if (mqttClient != NULL) {
     delete mqttClient;
```

Comment:
Is it safe to delete the object created by a user?
Maybe he destructs us but still needs his Internet connection.
Or am I miss the point?